pkg/rid/store/cockroach/store.go

```diff
@@ -48,33 +48,37 @@ type repo struct {
 // TODO: Add the SCD interfaces here, and collapse this store with the
 // outer pkg/cockroach
 type Store struct {
-	db     *cockroach.DB
-	logger *zap.Logger
-	clock  clockwork.Clock
+	db      *cockroach.DB
+	logger  *zap.Logger
+	clock   clockwork.Clock
+	version *semver.Version
 }
 
 // NewStore returns a Store instance connected to a cockroach instance via db.
 func NewStore(ctx context.Context, db *cockroach.DB, logger *zap.Logger) (*Store, error) {
+	vs, err := db.GetVersion(ctx, DatabaseName)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Failed to get database schema version for remote ID")
+	}
+
 	store := &Store{
-		db:     db,
-		logger: logger,
-		clock:  DefaultClock,
+		db:      db,
+		logger:  logger,
+		clock:   DefaultClock,
+		version: vs,
 	}
 
-	if err := store.CheckCurrentMajorSchemaVersion(ctx); err != nil {
+	if err := store.checkCurrentMajorSchemaVersion(*vs); err != nil {
 		return nil, stacktrace.Propagate(err, "Remote ID schema version check failed")
 	}
 
 	return store, nil
 }
 
 // CheckCurrentMajorSchemaVersion checks that store supports the current major schema version.
-func (s *Store) CheckCurrentMajorSchemaVersion(ctx context.Context) error {
-	vs, err := s.GetVersion(ctx)
-	if err != nil {
-		return stacktrace.Propagate(err, "Failed to get database schema version for remote ID")
-	}
-	if vs == cockroach.UnknownVersion {
+func (s *Store) checkCurrentMajorSchemaVersion(vs semver.Version) error {
```

Comment:
I don't think we need to pass in the version as a parameter right, since its already part of the store object when New Store is called, also I believe we want to keep this as public method as well to preserve the interface, as Thomas mentioned

Reply:
This method is not part of store interface and also no other class call it. Do we have up coming changes that use this method?